images_config/kickstart/kickstart-robot.sh

```diff
@@ -0,0 +1,465 @@
+        HOSTNAME=$(cat /etc/hostname)
+
+        SERVERNAME=$(echo ${HOSTNAME%-*}-b1)
+        echo $SERVERNAME
+        touch /etc/auto.direct
+        if grep -q "/u  -fstype=nfs4    $SERVERNAME:/" /etc/auto.direct && grep -q "/-  /etc/auto.direct" /etc/auto.master ; then
+            echo "NFS setup already in /etc/auto.direct or /etc/auto.master, skipping NFSSetup for slave"
+        else
+            echo "/u  -fstype=nfs4    $SERVERNAME:/" >> /etc/auto.direct
+            echo "/-  /etc/auto.direct" >> /etc/auto.master
+        fi
+        update-rc.d autofs defaults
+        service autofs restart
+        modprobe nfs
+    fi
+}
+
+function AddUsers {
+    printHeader "AddUsers"
+
+    #add robot-local user (preseedd seems to have a bug adding the user before postscript execution, results in overwritten group and user ids)
+    useradd -b /home -d /home/robot-local -m -s /bin/bash -k /etc/skel robot-local
+    echo 'robot-local:$1$.8rMo3Kc$hwkXrTTshYmLa9iplJchz.' | chpasswd -e
+
+    #Give robot-local full sudo rights
+    if grep -q "robot-local ALL=(ALL) NOPASSWD: ALL" /etc/sudoers ; then
+        echo "found robot-local NOPASSWD in sudoers already, skipping GiveFullRights to robot-local"
+    else
+        echo "robot-local ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
+    fi
+
+    if [ "$INSTALL_TYPE" == "master" ]; then      
+
+        #Add robot user
+        mkdir /u
+        mount --bind /home /u
+        useradd -b /u -d /u/robot -m -s /bin/bash -k /etc/skel robot
+        echo 'robot:$1$.8rMo3Kc$hwkXrTTshYmLa9iplJchz.' | chpasswd -e
+    fi
+    #Give robot user full rights for sudo
+    if grep -q "robot ALL=(ALL) NOPASSWD: ALL" /etc/sudoers ; then
+        echo "found robot NOPASSWD in sudoers already, skipping GiveFullRights to robot"
+    else
+        echo "robot ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
+    fi
+}
+
+function InstallROS {
+    printHeader "InstallROS"
+    if grep -q "deb http://packages.ros.org/ros/ubuntu $DISTRO main" /etc/apt/sources.list.d/ros-latest.list ; then
+        echo "Ros sources already setup. Skipping setup"
+    else
+        echo "deb http://packages.ros.org/ros/ubuntu $DISTRO main" > /etc/apt/sources.list.d/ros-latest.list
+        apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
+    fi
+
+    apt-get update
+    apt-get install ros-kinetic-ros-base -y
+    apt-get install python-rosinstall python-rosinstall-generator python-wstool -y
+    apt-get install python-catkin-tools -y
+    apt-get install python-pip -y
+}
+
+function SetupGrubRecFail {
+    printHeader "SetupGrubRecFail"
+    if grep -q GRUB_RECORDFAIL_TIMEOUT= /etc/default/grub ; then
+        echo "found GRUB_RECORD_FAIL flag already, skipping SetupGrubRecFail (update-grub call)"
+    else
+        echo "GRUB_RECORDFAIL_TIMEOUT=2" >> /etc/default/grub
+        update-grub
+    fi
+}
+
+#TODO: requires user input - No working solution found so far
```

Comment:
isn't this fixed by https://github.com/ipa320/setup_cob4/pull/199/files#diff-448827c30f63f80de9bfc13be7a5bb3eR41?
if so, remove this function

Reply:
if no user input is availabe, as it is on stick install, user input is not required for this to work.